src/include/daos_prop.h

```diff
@@ -387,16 +388,44 @@ daos_label_is_valid(const char *label)
 	if (len == 0 || len > DAOS_PROP_LABEL_MAX_LEN)
 		return false;
 
-	/** Verify that it contains only alphanumeric characters or :._ */
+	/** Verify that it contains only alphanumeric characters or :.-_ */
 	for (i = 0; i < len; i++) {
 		char c = label[i];
 
 		if (isalnum(c) || c == '.' || c == '_' || c == ':')
 			continue;
+		if (c == '-') {
+			maybe_uuid = true;
+			continue;
+		}
 
 		return false;
 	}
 
+	/** Check to see if it could be a valid UUID */
+	if (maybe_uuid && strnlen(label, 36) == 36) {
+		bool		is_uuid = true;
+		const char	*p;
+
+		/** Implement the check directly to avoid uuid_parse() overhead */
```

Comment:
is uuid_parse much more expensive to warrant doing this check?

Reply:
im asking because im using uuid_parse liberally in other places to check for proper uuid and maybe we should abstract this out and use this instead

Reply:
This function was rewritten by @johannlombardi after the original regex-based implementation was found to be stack hungry, so I wanted to avoid similar issues. There's a lot going on in uuid_parse() that we don't really need here: https://github.com/karelzak/util-linux/blob/master/libuuid/src/parse.c